app/src/ui/autocompletion/autocompleting-text-input.tsx

```diff
@@ -295,17 +286,28 @@ export abstract class AutocompletingTextInput<
     const autoCompleteText = autocompletionState.provider.getCompletionText(
       item
     )
+
+    const textWithAutoCompleteText =
+      originalText.substr(0, range.start - 1) + autoCompleteText + ' '
+
     const newText =
-      originalText.substr(0, range.start - 1) +
-      autoCompleteText +
-      originalText.substr(range.start + range.length) +
-      ' '
+      textWithAutoCompleteText + originalText.substr(range.start + range.length)
+
     element.value = newText
 
     if (this.props.onValueChanged) {
       this.props.onValueChanged(newText)
     }
 
+    // This is pretty gross. Clicking on the list moves focus off the text area.
+    // Immediately moving focus back doesn't work. Gotta wait a runloop I guess?
```

Comment:
This is already an issue with the current app, and I'm not sure of alternatives to fix this.

@niik any other ideas to handle clicking the autocomplete list item stealing focus from the related text element?

Reply:
Not of the top of my head, no. But we don't have to solve that in this PR. But it seems like this PR changes to doing the focus hack regardless of whether the autocomplete happened as the result of a keyboard or pointer action. We should _only_ do this onClick.

Reply:
@niik good shout. I'll see if I can work with @CarlRosell to handle keyboard autocomplete differently to pointer autocompletes :thumbsup: